packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -0,0 +1,438 @@
+            this.quadrantRefs[QuadrantType.TOP].scrollContainer.scrollLeft = nextScrollLeft;
+        }
+        this.props.onScroll(event);
+    }
+
+    // Size syncing
+    // ============
+
+    private syncQuadrantMenuElementWidths() {
+        this.syncQuadrantMenuElementWidth(QuadrantType.MAIN);
+        this.syncQuadrantMenuElementWidth(QuadrantType.TOP);
+        this.syncQuadrantMenuElementWidth(QuadrantType.LEFT);
+        this.syncQuadrantMenuElementWidth(QuadrantType.TOP_LEFT);
+    }
+
+    private syncQuadrantMenuElementWidth(quadrantType: QuadrantType) {
+        const mainQuadrantMenu = this.quadrantRefs[QuadrantType.MAIN].menu;
+        const mainQuadrantRowHeader = this.quadrantRefs[QuadrantType.MAIN].rowHeader;
+        const quadrantMenu = this.quadrantRefs[quadrantType].menu;
+
+        // the main quadrant menu informs the size of every other quadrant menu
+        if (mainQuadrantMenu != null && mainQuadrantRowHeader != null && quadrantMenu != null) {
+            const { width } = mainQuadrantRowHeader.getBoundingClientRect();
+            quadrantMenu.style.width = `${width}px`;
+
+            // no need to use the main quadrant's menu to set its *own* height
+            if (quadrantType !== QuadrantType.MAIN) {
+                const { height } = mainQuadrantMenu.getBoundingClientRect();
+                quadrantMenu.style.height = `${height}px`;
+            }
+        }
+    }
+
+    private syncQuadrantSizes() {
+        const mainQuadrantScrollElement = this.quadrantRefs[QuadrantType.MAIN].scrollContainer;
+        const topQuadrantElement = this.quadrantRefs[QuadrantType.TOP].quadrant;
+        const topQuadrantRowHeaderElement = this.quadrantRefs[QuadrantType.TOP].rowHeader;
+        const leftQuadrantElement = this.quadrantRefs[QuadrantType.LEFT].quadrant;
+        const topLeftQuadrantElement = this.quadrantRefs[QuadrantType.TOP_LEFT].quadrant;
+        const topLeftQuadrantRowHeaderElement = this.quadrantRefs[QuadrantType.TOP_LEFT].rowHeader;
+
+        const { grid, numFrozenColumns, numFrozenRows } = this.props;
+
+        // if there are no frozen rows or columns, we still want the quadrant to be 1px bigger to
+        // reveal the header border.
+        const BORDER_WIDTH_CORRECTION = 1;
+
+        const leftQuadrantGridContentWidth = numFrozenColumns > 0
+            ? grid.getCumulativeWidthAt(numFrozenColumns - 1)
+            : BORDER_WIDTH_CORRECTION;
+        const topQuadrantGridContentHeight = numFrozenRows > 0
+            ? grid.getCumulativeHeightAt(numFrozenRows - 1)
+            : BORDER_WIDTH_CORRECTION;
+
+        // all menus are the same size, so arbitrarily use the one from the main quadrant.
+        // assumes that the menu element width has already been sync'd after the last render
+
+        const rowHeader = this.findRowHeader(QuadrantType.MAIN);
+        const columnHeader = this.findColumnHeader(QuadrantType.MAIN);
```

Comment:
depends how often this component updates... if it's every scroll event then yes that's too expensive. if not, then we're probably okay. worth gathering some data.